arborio/include/arborio/cableio.hpp

```diff
@@ -0,0 +1,39 @@
+#pragma once
+
+#include <arbor/cable_cell.hpp>
+#include <arbor/s_expr.hpp>
+
+#define CABLE_CELL_FORMAT_VERSION 1
+
+namespace arborio {
+
+struct cableio_parse_error: arb::arbor_exception {
+    explicit cableio_parse_error(const std::string& msg, const arb::src_location& loc);
+};
+
+struct cableio_morphology_error: arb::arbor_exception {
+    explicit cableio_morphology_error(const unsigned bid);
+};
+
+template <typename T>
+using parse_hopefully = arb::util::expected<T, cableio_parse_error>;
+using cable_cell_variant = std::variant<arb::morphology, arb::label_dict, arb::decor, arb::cable_cell>;
+
+struct meta_data {
+    int version = CABLE_CELL_FORMAT_VERSION;
+};
+struct cable_cell_component {
+    meta_data meta;
+    cable_cell_variant component;
+};
+
+std::ostream& write_component(std::ostream&, const cable_cell_component&);
+std::ostream& write_component(std::ostream&, const arb::decor& x, const meta_data& m = {});
+std::ostream& write_component(std::ostream&, const arb::label_dict& x, const meta_data& m = {});
+std::ostream& write_component(std::ostream&, const arb::morphology& x, const meta_data& m = {});
+std::ostream& write_component(std::ostream&, const arb::cable_cell& x, const meta_data& m = {});
+
+parse_hopefully<cable_cell_component> parse_component(const std::string&);
```

Comment:
I have a suggestion on the readers here. Maybe it's wise to allow to pass an expected version or accept a range of compatible versions. The reasoning is that users might have older files they wish to read in. Storing should only happen in the latest format, however.

Reply:
The string argument is expected to contain meta-data containing the version. At the moment only version 1 is supported, and anything else will return an error. Is this what you meant, or was it something else?

Reply:
Not exactly. Assume someone writes out a cable-cell on version 1 of the format, then upgrades to a version of arbor using version 2. Now they try reading the old file. What are we doing then. Here's the options I see

1. Flat out refuse to load and throw `unsupported_version_error`.
2. Like 1. but offer also a script to convert version 1 -> 2.
3. Support more than one version and automatically select the loader.
4. Like 3., but the load function takes a version, which is our expected version.

I think it's worth to lay the foundation for this now.

Reply:
Right now, we do option (1). We could do one of the other options in the future once we support more versions. 
Personally, I like either option (2) or (3), but I haven't thought of it too deeply.

Reply:
Ok, that potentially entails changing the public API, if going with 4. We can leave it is for this, as long we consider this be acceptable.